source/agora/consensus/state/UTXOCache.d

```diff
@@ -257,4 +280,37 @@ public class TestUTXOSet : UTXOCache
     {
         this.storage[utxo] = value;
     }
+
+    /***************************************************************************
+
+        Returns:
+            the number of elements in the UTXO set
+
+    ***************************************************************************/
+
+    public override size_t length () @safe
+    {
+        return this.storage.length;
+    }
+
+    /***************************************************************************
+
+        Get UTXOs from the UTXO set
+
+        Params:
+            pubkey = the key by which to search UTXOs in UTXOSet
+
+        Returns:
+            the associative array for UTXOs
+
+    ***************************************************************************/
+
+    public override UTXO[Hash] getUTXOs (const ref PublicKey pubkey) nothrow @safe
+    {
+        UTXO[Hash] utxos;
+        foreach(const ref key_val; storage.byKeyValue)
```

Comment:
```suggestion
        foreach (const ref key, const ref val; storage)
```

D is smart, no need for `byKeyValue`

Reply:
I think he used it because `_aaApply2 is not nothrow`.

Reply:
> I think he used it because `_aaApply2 is not nothrow`.

exactly....

Reply:
Just pretend I didn't say "D is smart".